cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -177,14 +182,14 @@ public void addWorkflowNodeState(ProgramRunId workflowRunId, WorkflowNodeStateDe
     write(key, nodeStateDetail);
   }
 
-  private void addWorkflowNodeState(Id.Program program, String pid, Map<String, String> systemArgs,
+  private void addWorkflowNodeState(ProgramId programId, String pid, Map<String, String> systemArgs,
                                     ProgramRunStatus status, @Nullable Throwable failureCause) {
     String workflowNodeId = systemArgs.get(ProgramOptionConstants.WORKFLOW_NODE_ID);
     String workflowName = systemArgs.get(ProgramOptionConstants.WORKFLOW_NAME);
     String workflowRun = systemArgs.get(ProgramOptionConstants.WORKFLOW_RUN_ID);
 
-    ProgramRunId workflowRunId = new ProgramRunId(program.getNamespaceId(), program.getApplicationId(),
-                                                  ProgramType.WORKFLOW, workflowName, workflowRun);
+    ProgramRunId workflowRunId = Ids.namespace(programId.getNamespace()).app(programId.getApplication())
```

Comment:
This is ProgramRunId for the workflow while the input is program id of the program which Workflow started.